platform/posix/transport/utest/openssl_utest.c

```diff
@@ -0,0 +1,773 @@
+        returnStatus = Openssl_Connect( &networkContext,
+                                        &serverInfo,
+                                        &opensslCredentials,
+                                        SEND_RECV_TIMEOUT,
+                                        SEND_RECV_TIMEOUT );
+        TEST_ASSERT_EQUAL( expectedStatus, returnStatus );
+    }
+}
+
+/**
+ * @brief Cover cases in which TLS credentials and configuration parameters are
+ * set to NULL.
+ */
+void test_Openssl_Connect_NULL_Members_In_Creds_And_Configs( void )
+{
+    OpensslStatus_t returnStatus, expectedStatus = OPENSSL_INVALID_CREDENTIALS;
+
+    /* Coverage for NULL pRootCaPath. */
+    memset( &opensslCredentials, 0, sizeof( OpensslCredentials_t ) );
+    /*expectedStatus = failFunctionFrom_Openssl_Connect( SSL_get_verify_result_fn, NULL );*/
+    Sockets_Connect_ExpectAnyArgsAndReturn( SOCKETS_SUCCESS );
+    TLS_client_method_ExpectAndReturn( &sslMethod );
+    SSL_CTX_new_ExpectAnyArgsAndReturn( &sslCtx );
+    SSL_CTX_free_ExpectAnyArgs();
+    returnStatus = Openssl_Connect( &networkContext,
+                                    &serverInfo,
+                                    &opensslCredentials,
+                                    SEND_RECV_TIMEOUT,
+                                    SEND_RECV_TIMEOUT );
+    TEST_ASSERT_EQUAL( expectedStatus, returnStatus );
+
+    /* Coverage for NULL pClientCertPath, pPrivateKeyPath, pAlpnProtos, sniHostName,
+     * and maxFragmentLength == 0. */
+    opensslCredentials.pRootCaPath = ROOT_CA_CERT_PATH;
+    expectedStatus = failFunctionFrom_Openssl_Connect( SSL_get_verify_result_fn, NULL );
+    returnStatus = Openssl_Connect( &networkContext,
+                                    &serverInfo,
+                                    &opensslCredentials,
+                                    SEND_RECV_TIMEOUT,
+                                    SEND_RECV_TIMEOUT );
+    TEST_ASSERT_EQUAL( expectedStatus, returnStatus );
+
+    /* Coverage for alpnProtosLen == 0. */
+    opensslCredentials.pAlpnProtos = ALPN_PROTOS;
+    expectedStatus = failFunctionFrom_Openssl_Connect( SSL_get_verify_result_fn,
+                                                       NULL );
+    returnStatus = Openssl_Connect( &networkContext,
+                                    &serverInfo,
+                                    &opensslCredentials,
+                                    SEND_RECV_TIMEOUT,
+                                    SEND_RECV_TIMEOUT );
+    TEST_ASSERT_EQUAL( expectedStatus, returnStatus );
+}
+
+/**
+ * @brief Test the happy path case in which the TLS connection is established.
+ */
+void test_Openssl_Connect_Succeeds( void )
+{
+    OpensslStatus_t returnStatus, expectedStatus;
+
+    /* Set the parameter to one more that the max enum value of the function names
+     * so that no functions fail for this happy path case. */
+    expectedStatus = failFunctionFrom_Openssl_Connect( SSL_get_verify_result_fn + 1,
```

Comment:
is this required? can't we just assign the success status?

Reply:
Still need it to expect the openssl API calls, but I will void return value change assert to compare to `OPENSSL_SUCCESS` for readability.